telosnetwork/telos-wallet
# #420 | Deeplink handling

# Fixes #420 

## Description
This PR does a couple of things
1. users who are not logged in are now redirected to the homepage when attempting to access a route that requires auth
2. enable some pages (demo, nft details) to be viewed while not logged in
3. when viewing a page while not logged in, adds the ability to sign in using the left nav
4. slightly changes the logic for when to display the 'back' button in header to be simpler

## Test scenarios
- go to https://deploy-preview-421--wallet-staging.netlify.app/
    - make sure you are logged out / do not log in
- go to the following pages: [send](https://deploy-preview-421--wallet-staging.netlify.app/evm/send), [receive](https://deploy-preview-421--wallet-staging.netlify.app/evm/receive), [wallet](https://deploy-preview-421--wallet-staging.netlify.app/evm/wallet), [NFT inventory](https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-inventory)
    - you should be redirected home page
- go to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
    - you should be able to see the NFT details page
    - you should see 'Log in' on the nav menu rather than Log Out
- click wallet or collectibles in the nav
    - you should be directed home
- go back to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
- click log in in the nav
    - the login modal should pop
- log in using metamask
    -  you should see a success message
    - the nav should change to say 'log out'
    - you should be able to freely navigate around the site
- log out
- go back to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
- click log in
- log in using wallet connect
    -  you should see a success message
    - the nav should change to say 'log out'
    - you should be able to freely navigate around the site

## Checklist:
